include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp

```diff
@@ -133,6 +134,17 @@ static Value getTransposedValue(Value source, const Location loc,
   return transpose;
 }
 
+// for IntLike and FloatLike types
+static std::optional<unsigned> getBitWidth(Type a) {
+  if (auto type = dyn_cast<TensorType>(a))
+    return type.getElementType().getIntOrFloatBitWidth();
+
```

Comment:
i think we're still missing a check if the element type is indeed int or float

Reply:
right, should be fixed now, thank you!